### PR TITLE
[BB-2961] [BD-05] [TNL-7577] Added a templating mixin

### PR DESCRIPTION
**Pull request by farhaanbukhsh**
This PR helps to interact with Studio API to give a list of presets

**JIRA**: [TNL-7577](https://openedx.atlassian.net/browse/TNL-7577)

**Discussions**: https://openedx.atlassian.net/browse/TNL-7577

**Dependencies**: https://github.com/edx/edx-platform/pull/25012

**Sandbox URL**: [Sandbox](https://studio.pr25012.sandbox.opencraft.hosting/) 

**Testing instructions**:

You should have a working devstack for this

1. Clone this repo in ~/dev_root/src
2.  Dev up lms_studio by `make dev.up.lms+studio`
3. Run:

```
docker exec -t edx.devstack.lms bash -c '. /edx/app/edxapp/venvs/edxapp/bin/activate && cd /edx/app/edxapp/edx-platform && pip uninstall -y ora2 && pip install -e /edx/src/edx-ora2 && pip freeze | grep ora2'


docker exec -t edx.devstack.studio bash -c '. /edx/app/edxapp/venvs/edxapp/bin/activate && cd /edx/app/edxapp/edx-platform && pip uninstall -y ora2 && pip install -e /edx/src/edx-ora2 && pip freeze | grep ora2'


```

This is for master branch but docker name might vary depending on which branch you are using.

4. Sign In to `Studio` edit a course and unit now when you click on `Open Response` button you will see a pop-up
5. The other pop-is are dummy but the first-option works.

**Author notes and concerns**:

~The presets are somewhat a dummy yet, but mostly the inspiration is taken from how `Problems` and `HTML` module~
~works, specially this https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/x_module.py#L1037~
~class and it used in HTML~

~https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/html_module.py#L49~

~I was not sure about the translation running using django and it's integration with Studio, couldn't get that
to work.~

**Screenshot**:
image

image

**Reviewers**
- [ ] @giovannicimolin
- [ ] edX reviewer[s] TBD

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @farhaanbukhsh! I've created [BLENDED-595](https://openedx.atlassian.net/browse/BLENDED-595) to keep track of it in Jira.   More details are on the [BD-05](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1529970851/BD-05+ORA+v3+Usability+Moderation) project page.

When this pull request is ready, tag your edX technical lead.

File: openassessment/xblock/openassessmentblock.py [hunk 1]
```diff
@@ -127,6 +129,13 @@ class OpenAssessmentBlock(MessageMixin,
         'staff-assessment',
     ]
 
+    VALID_ASSESSMENT_TYPES_DISPLAY_NAMES = {
```
Comment (giovannicimolin) [inline openassessment/xblock/openassessmentblock.py:132]: Can you add the types from the mockups and give them more descriptive names here?

File: openassessment/xblock/openassesment_template_mixin.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    This helps to get templates for different type of assessment that is
+    offered.
+    """
+
+    VALID_ASSESSMENT_TYPES_DISPLAY_NAMES = {
+        "peer-assessment": _("Peer Assessment Only"),
+        "self-assessment": _("Self Assessment Only"),
+        "staff-assessment": _("Staff Assessment Only"),
+        "self-to-peer": _("Self Assessment to Peer Assessment"),
+        "self-to-staff": _("Self Assessment to Peer Assessment")
```
Comment (giovannicimolin) [inline openassessment/xblock/openassesment_template_mixin.py:18]:
```suggestion
        "self-to-staff": _("Self Assessment to Staff Assessment")
```

**Review by giovannicimolin — CHANGES_REQUESTED** [review]
Changes look good, and seem to be working just fine in the devstack.

A few things to improve:
- The item ordering is not as in the mockups:
image
- We should change from `Open Response Assessment` to `Basic Open Response Assessment Template` as in the mockups.
- Fix quality issues (check CI run for details)
- Run `make extract_translations` and commit the result.

**Comment by farhaanbukhsh** [comment]
> Changes look good, and seem to be working just fine in the devstack.
> 
> A few things to improve:
> 
> * The item ordering is not as in the mockups:
>   image

Done 

> * We should change from `Open Response Assessment` to `Basic Open Response Assessment Template` as in the mockups.

This can be done on studio side https://github.com/edx/edx-platform/blob/f7ddced9f83191ebcdcd736a22583ed020b10993/cms/djangoapps/contentstore/views/component.py#L294
at this line we can pass a custom name . This actually is taking the name of the xblock.

> * Fix quality issues (check CI run for details)

Done 

> * Run `make extract_translations` and commit the result.

Done


Hey @giovannicimolin  I have addressed your comments :) Please let me know.

**Comment by giovannicimolin** [comment]
@farhaanbukhsh The item ordering is still not fixed. Can you follow the mockups linked in the ticket? 
The image I've posted is a screenshot showing the wrong positioning.

**Review by giovannicimolin — APPROVED** [review]
:+1:

- [x] I tested this:
1. Checked out this branch and installed on devstack.
2. Got into Studio, create a course.
3. Clicked on `Open Responses` button (I'm also using https://github.com/edx/edx-platform/pull/25012).
4. Checked that the templates described in the mockup are rendered.
image
5. Checked that each template correctly configures the ORA block.
- [x] I read through the code
- [x] I checked for accessibility issues: Basic checks on items added.
- [ ] Includes documentation: No :x:. We need to update the ORA documentation later.

@farhaanbukhsh I think this is enough for changes in this repo, though we don't exactly match 1:1 the mockups.

A few requests **before** asking for upstream review:
1. Update the PR description to use our template.
2. Add screenshots.

**Comment by farhaanbukhsh** [comment]
> @farhaanbukhsh The item ordering is still not fixed. Can you follow the mockups linked in the ticket?
> The image I've posted is a screenshot showing the wrong positioning. It might be local to my setup, can you post a screenshot?

You are right I thought it would have fixed it anyhow I investigated and there doesn't seem to be a way to keep the order intact since it gets converted into a string at the UI level.

https://github.com/edx/edx-platform/blob/master/cms/templates/container.html#L45

so I don't know if we can have a control over it.

**Review by jnlapierre — APPROVED** [review]
LGTM

**Comment by giovannicimolin** [comment]
@farhaanbukhsh Can you fix the merge conflicts here?

**Comment by farhaanbukhsh** [comment]
> @farhaanbukhsh Can you fix the merge conflicts here?

Done 😄

**Pull request merged**